### ISSUE TITLE
Issue: currentRoute is blank on page load

### ISSUE THREAD

**Issue by fabien** [easyroute-router/svelte-easyroute#26]
Hi, first off, this router looks really promising! Nice work!

While playing around with it in a sandbox, I noticed that `currentRoute` is never populated on the initial load. It is only set when switching to another page - is this a bug? Please see the console at:

https://codesandbox.io/s/svelte-easyroute-simpliest-demo-forked-2k18c

One more thing, the `beforeEach` doesn't seem to work properly either, the `fullPath` is always identical on both `from` and `to`.

Another thing that I noticed is that apparently, `useCurrentRoute` currently doesn't return a plain readable Svelte store. If it did, it would make the boilerplate for each component a bit more compact (using reactive `$` code), and you would not have to unsubscribe manually. Any thoughts on this?

**Comment by fabien** [comment]
The `beforeEach` bug seems to have been introduced in the `3.0.1` release. Version `3.0.1` works as expected. However, `currentRoute` is not actually set on initial load here either.

**Comment by lyohaplotinka** [comment]
Hello @fabien! 

- Issue with empty `currentRoute` object on startup is fixed, however, I decided to declare it deprecated and the documentation now recommends using the `useCurrentRoute` hook since it can work not only in direct children of RouterOutlet.
- A bug with navigation hooks has also been fixed. To avoid misunderstandings, I note that in your example, in the hook that displays the paths to the console, the "to" and "from" places are reversed (in the console.log)

As for useCurrentRoute, it is not intended to return a valid Svelte store, as it refers to an Observable implemented internally by easyroute-core. It simply provides access to its value.

However, I like the idea of returning the svelte-repository, and perhaps it will be implemented. But not as part of the work on this issue.

In version 3.0.7, the described errors have been fixed.
